File: ANDROID/ActivityLiveCycle/app/src/main/java/com/aramzy/activitylivecycle/MainActivity.java
package com.aramzy.activitylivecycle;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = MainActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.i(TAG, "LOG: onCreate");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.i(TAG, "LOG: onStart");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i(TAG, "LOG: onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.i(TAG, "LOG: onPause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.i(TAG, "LOG: onStop");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.i(TAG, "LOG: onRestart");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.i(TAG, "LOG: onDestroy");
    }

    public void openSettings(View view) {
        Intent intent = new Intent(this, SettingsActivity.class);
        startActivity(intent);
    }

    public void openDialog(View view) {
        Intent intent = new Intent(this, DialogActivity.class);
        startActivity(intent);
    }
}

File: ANDROID/MyfireAuth/app/src/main/java/com/aramzy/myfireauth/MainActivity.kt
package com.aramzy.myfireauth

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import com.google.android.gms.tasks.OnCompleteListener
import com.google.firebase.auth.FirebaseAuth

class MainActivity : AppCompatActivity() {

    var myAuth = FirebaseAuth.getInstance()

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.button).setOnClickListener(View.OnClickListener {

            if(findViewById<EditText>(R.id.editText).text.toString() != "" && findViewById<EditText>(R.id.editText2).text.toString() != "") {

                signUp(findViewById<EditText>(R.id.editText).text.toString().trim(), findViewById<EditText>(R.id.editText2).text.toString().trim())

            } else {

                Toast.makeText(this, "Please fill all fields", Toast.LENGTH_LONG).show()

            }

        })

        findViewById<Button>(R.id.button2).setOnClickListener(View.OnClickListener {

            view -> signIn(view, findViewById<EditText>(R.id.editText).text.toString().trim(), findViewById<EditText>(R.id.editText2).text.toString().trim())

            findViewById<EditText>(R.id.editText).setText("")
            findViewById<EditText>(R.id.editText2).setText("")
            findViewById<EditText>(R.id.editText).requestFocus()

        })

    }

    private fun signUp(mail: String, pass: String) {

        myAuth.createUserWithEmailAndPassword(mail, pass).addOnCompleteListener(this, OnCompleteListener {

            task -> if(task.isSuccessful) {Toast.makeText(this, "Done successfully", Toast.LENGTH_SHORT).show()} else {Toast.makeText(this, "${task.exception!!.message}", Toast.LENGTH_SHORT).show()}

        })

        findViewById<EditText>(R.id.editText).setText("")
        findViewById<EditText>(R.id.editText2).setText("")
        findViewById<EditText>(R.id.editText).requestFocus()

    }

    private fun signIn(view: View, mail: String, pass: String) {

        Toast.makeText(this, "Authenticating ... ", Toast.LENGTH_SHORT).show()

        myAuth.signInWithEmailAndPassword(mail, pass).addOnCompleteListener(this, OnCompleteListener {

            task -> if(task.isSuccessful) {

            val intent: Intent = Intent(this, LoginActivity::class.java)
            intent.putExtra("id", myAuth.currentUser!!.email)
            startActivity(intent)

        } else {

            Toast.makeText(this, "${task.exception!!.message}", Toast.LENGTH_SHORT).show()

        }

        })

    }

}